apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+              },
+            ],
+          },
+        ],
+      },
+      {
+        type: "callout",
+        content: {
+          type: "prose",
+          content: [
+            {
+              type: "paragraph",
+              content: [
+                {
+                  type: "text",
+                  text: "Test Callout content",
+                },
+              ],
+            },
+          ],
+        },
+      },
+    ]
+
+    type Page = Awaited<ReturnType<typeof setupPageResource>>["page"]
+    let pageToUpdate: Page
+    type UpdatePageOutput = z.output<typeof updatePageBlobSchema>
+    const createPageUpdateArgs = (page: Page) => {
+      return {
+        pageId: Number(page.id),
+        siteId: page.siteId,
+        content: JSON.stringify({
+          content: NEW_PAGE_BLOCKS,
+          layout: "content",
+          page: pick(page, ["title", "permalink"]),
+          version: "0.1.0",
+        } as UpdatePageOutput["content"]),
+      }
+    }
+
+    beforeEach(async () => {
+      const { page } = await setupPageResource({ resourceType: "Page" })
+      pageToUpdate = page
+    })
+
+    it("should throw 401 if not logged in", async () => {
+      const unauthedSession = applySession()
+      const unauthedCaller = createCaller(createMockRequest(unauthedSession))
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+      const result = unauthedCaller.updatePageBlob(pageUpdateArgs)
+
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should return 404 if page does not exist", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = caller.updatePageBlob({
+        ...pageUpdateArgs,
+        pageId: 999999, // should not exist
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "Resource not found",
+        }),
+      )
+    })
+
+    it("should return 422 if content is not valid", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = caller.updatePageBlob({
+        ...pageUpdateArgs,
+        content: "do not match the shape",
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError("Invalid page content")
+    })
+
+    it("should update draft page blob if args are valid and has current draft", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = await caller.updatePageBlob(pageUpdateArgs)
+
+      // Assert
+      const actual = await db
+        .selectFrom("Blob")
+        .where("id", "=", pageToUpdate.draftBlobId)
+        .select("content")
+        .executeTakeFirstOrThrow()
+      expect(actual.content).toEqual(result.content)
+    })
```

Comment:
@harishv7 trouble you to relook again at the draft logic. we should aim to have our invariants verified through test